File: shared/src/main/java/com/walter/lifelog/shared/dto/ImageResource.java
package com.walter.lifelog.shared.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import java.io.InputStream;

@Schema(description = "이미지 리소스")
public record ImageResource(
    @Schema(description = "이미지 InputStream", hidden = true)
    InputStream inputStream,
    @Schema(description = "MIME 타입", example = "image/jpeg")
    String mimeType,
    @Schema(description = "파일 이름", example = "photo.jpg")
    String fileName,
    @Schema(description = "파일 크기 (bytes)", example = "2048576")
    long fileSize
) {}

File: photo-archive-service/src/main/kotlin/com/walter/lifelog/photo/dto/ImageResource.kt
package com.walter.lifelog.photo.dto

import io.swagger.v3.oas.annotations.media.Schema
import java.io.InputStream

@Schema(description = "이미지 리소스")
data class ImageResource(
    @Schema(description = "이미지 InputStream", hidden = true)
    val inputStream: InputStream,

    @Schema(description = "MIME 타입", example = "image/jpeg")
    val mimeType: String,

    @Schema(description = "파일 이름", example = "photo.jpg")
    val fileName: String,

    @Schema(description = "파일 크기 (bytes)", example = "2048576")
    val fileSize: Long
) {
    fun isImage(): Boolean = mimeType.startsWith("image/")
}

